operator/v1/types_etcd.go

```diff
@@ -43,6 +43,16 @@ type EtcdSpec struct {
 	// +openshift:enable:FeatureGate=HardwareSpeed
 	// +optional
 	HardwareSpeed ControlPlaneHardwareSpeed `json:"controlPlaneHardwareSpeed"`
+
+	// BackendQuotaGiB sets the etcd backend storage size limit in gibibytes. Defaults to 8GiB.
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:default:=8
+	// +kubebuilder:validation:Maximum=32
+	// +kubebuilder:validation:XValidation:rule="self>=oldSelf",message="can't decrease database size"
+	// +openshift:enable:FeatureGate=EtcdBackendQuota
+	// +default:=8
+	// +optional
+	BackendQuotaGiB int64 `json:"backendQuotaGiB,omitempty"`
```

Comment:
Does this really need to be an `int64`, given the small nature of this number, `int32` would be sufficient

Reply:
we have discussed this in the [enhancement](https://github.com/openshift/enhancements/pull/1549) 

everyone agreed upon `int64`

Reply:
Can you link to a specific thread? I can see conversation about resource.Quantity but not saying that it should be `int64` over `int32`. Why choose a larger storage size when the range will always be a small integer?

Reply:
This should not have been resolved, still not sure why this needs to be an int64

Reply:
So the field in etcd upstream is uin64 because it is the storage size in bytes

I initially followed the upstream convention - but the enhancement reviews recommeded ( to be user friendly ) to let the field being set as regular `int` while doing the conversion in the reconceilation logic

I actually agree to change this but has to be changed on the enhacement first

Reply:
I'd suggest getting the API right here and then updating the EP, it's minor enough

Reply:
@JoelSpeed can u inline your comment here please https://github.com/openshift/enhancements/pull/1549 ?

Reply:
> Does this really need to be an int64, given the small nature of this number, int32 would be sufficient
wdyt @tjungblu @dusk125 @hasbro17 @soltysh @deads2k 

I actually agree with @JoelSpeed it should be `int32` since it the value will never exceed this .. in fact, `int8` is sufficient :D